python/lsst/dax/distribution/chunklogs.py

```diff
@@ -0,0 +1,413 @@
+#!/usr/bin/env python3
+
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+
+class ChunkListFile:
+    """Read and write a set of chunks to a file.
+    """
+
+    def __init__(self, fname):
+        self._fname = fname
+        if self._fname:
+            self._fname = os.path.expanduser(self._fname)
+            self._fname = os.path.abspath(self._fname)
+        self.file_wopen = False
+        self.chunk_set = set()
+
+    def read(self):
+        with open(self._fname, 'r') as list_file:
+            f_raw = list_file.read()
+        self.parse(f_raw)
+
+    def parse(self, raw, separator='\n'):
+        """Parse the raw string for chunk numbers to put in chunk_set.
+
+        Parameters
+        ----------
+        raw : str
+            String to parse for chunk id numbers.
+        separator : str optional
+            Character or string used to separate id numbers. The normal
+            separator is '\n' but ',' is more appropriate for strings
+            entered on the command line.
+
+        Note
+        ----
+        Extra separators are ignored.
+        """
+        split_raw = raw.split(separator)
+        for st in split_raw:
+            if ':' in st:
+                st_split = st.split(':')
+                if len(st_split) == 2:
+                    val_a = int(st_split[0])
+                    val_b = int(st_split[1])
+                    if val_a > val_b:
+                        tmp = val_a
+                        val_a = val_b
+                        val_b = tmp
+                    for j in range(val_a, val_b + 1):
+                        self.chunk_set.add(j)
+                else:
+                    raise ValueError(f"value error in st={st} {st_split}")
```

Comment:
Can we try to reduce the nesting here?  For example, check != 2 for ValueError() first, then rest can be flattened one level up.